.github/workflows/release.yml

```diff
@@ -0,0 +1,32 @@
+name: Release to packagist
+
+on:
+  push:
+    tags:
+      - "*"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - id: get_data
+        run: echo "approvers=$(cat .github/CODEOWNERS | grep @ | tr -d '* ' | sed 's/@/,/g' | sed 's/,//1')" >> $GITHUB_OUTPUT
+
+      - uses: trstringer/manual-approval@v1
+        with:
+          secret: ${{ github.TOKEN }}
+          approvers: ${{ steps.get_data.outputs.approvers }}
+          minimum-approvals: 2
+          issue-title: 'Release opensearch-PHP'
+          issue-body: "Please approve or deny the release of opensearch-PHP client on packagist. **TAG**: ${{ github.ref_name }}  **COMMIT**: ${{ github.sha }}"
+          exclude-workflow-initiator-as-approver: true
+
+      - name: Release to Packagist
+        run: curl -XPOST -f -H 'content-type:application/json' 'https://packagist.org/api/update-package?username=opensearch&apiToken=${{secrets.PACKAGIST_PUBLISHING_API_TOKEN}}' -d'{"repository":{"url":"https://github.com/opensearch-project/opensearch-php"}}'
```

Comment:
Can we maybe just set up it as a webhook on the GitHub Repository itself? 

Here is an example of a project of mine
<img width="1314" alt="image" src="https://github.com/opensearch-project/opensearch-php/assets/6224096/aa24f4b2-a3eb-444d-9c1c-e0356e26ec16">

It's pretty valueable, when the branches are also keeped up to date on packagist.

Reply:
I believe other maintainer wanted to only publish based on tag https://github.com/opensearch-project/opensearch-build/issues/3146#issuecomment-1551874485
@dblock @VachaShah @harshavamsi Asking for your input here

Reply:
@shyim Want to open a separate issue for this? I am concerned that there might be security implications of auto-running workflows on merge.